src/cpp/src/llm_pipeline.cpp

```diff
@@ -334,7 +335,139 @@ std::pair<std::string, Any> generation_config(const GenerationConfig& config) {
 }  // namespace genai
 }  // namespace ov
 
-using namespace std;
+namespace {
+using namespace ov::genai;
+
+template<class... Ts> struct overloaded : Ts... {using Ts::operator()...;};
+template<class... Ts> overloaded(Ts...) -> overloaded<Ts...>;
+
+Tokenizer dont_construct() {
+    OPENVINO_THROW("Continuous Batching backend can't be constructed"
+        "from ireq because the model must be transformed");
+}
+
+class ContinuousBatchingAdapter final : public LLMPipelineImplBase {
+public:
+    ContinuousBatchingPipeline m_impl;
+
+    ContinuousBatchingAdapter(
+        const ov::InferRequest& request,
+        const Tokenizer& tokenizer,
+        OptionalGenerationConfig generation_config
+    ): LLMPipelineImplBase{dont_construct()}, m_impl{"", {}} {}
```

Comment:
It's the most advanced constructor which allows to fully customize a language model before LLMPipeline generation. The intention was to make it as generic as possible to avoid limiting possibilities.